internal/locate/region_request.go

```diff
@@ -322,7 +322,19 @@ type accessKnownLeader struct {
 
 func (state *accessKnownLeader) next(bo *retry.Backoffer, selector *replicaSelector) (*RPCContext, error) {
 	leader := selector.replicas[state.leaderIdx]
-	if leader.isExhausted(maxReplicaAttempt) {
+	liveness := reachable
+	if atomic.LoadInt32(&leader.store.unreachable) != 0 {
+		liveness = selector.checkLiveness(bo, leader)
+	}
+	if liveness == unreachable {
+		selector.state = &tryNewProxy{leaderIdx: state.leaderIdx}
```

Comment:
Why `tryNewProxy` here? I think it should try followers.

Reply:
Sorry, I missed the forwarding condition. The corresponding test is also fixed.